docs/docs/dev/implementing.md

```diff
@@ -1,22 +1,36 @@
+To do this, you will need to implement three interfaces from the `jelly-core` module: `ProtoEncoderConverter`, `ProtoDecoderConverter`, and `JellyConverterFactory`.
 
 - **ProtoEncoderConverter (serialization)**
-    - `get*` methods deconstruct triple and quad statements.
     - `nodeToProto` and `graphToProto` should translate into Jelly's representation all possible variations of RDF terms in the SPO and G positions, respectively.
-    - Example implementation for Jena: [JenaEncoderConverter]({{ git_link('jena/src/main/scala/eu/ostrzyciel/jelly/convert/jena/JenaEncoderConverter.scala') }})
-    - You can skip implementing this trait if you don't need serialization.
+    - Example implementation for Jena: [JenaEncoderConverter]({{ git_link('jena/src/main/java/eu/neverblink/jelly/convert/jena/JenaEncoderConverter.java') }})
+    - You can skip implementing this interface if you don't need serialization.
     - You can also skip implementing some methods (make them throw an exception or return null) if, for example, you don't want to work with quads.
 
 - **ProtoDecoderConverter (deserialization)**
-    - The `make*` methods should construct new RDF terms and statements. You can make them `inline`.
-    - Example implementation for Jena: [JenaDecoderConverter]({{ git_link('jena/src/main/scala/eu/ostrzyciel/jelly/convert/jena/JenaDecoderConverter.scala') }})
-    - You can skip implementing this trait if you don't need deserialization.
+    - The `make*` methods should construct new RDF terms and statements.
+    - Example implementation for Jena: [JenaDecoderConverter]({{ git_link('jena/src/main/java/eu/neverblink/jelly/convert/jena/JenaDecoderConverter.java') }})
+    - You can skip implementing this interface if you don't need deserialization.
     - You can also skip implementing some methods (make them throw an exception or return null) if, for example, you don't want to work with quads or RDF-star.
 
-- **ConverterFactory** – wrapper that allows other modules to use your converter.
+- **JellyConverterFactory** – wrapper that allows other modules to use your converter.
     - The methods should just return new instances (or singletons, if appropriate for your use case) of your `ProtoEncoderConverter` and `ProtoDecoderConverter` implementations.
-    - Example for Jena: [JenaConverterFactory]({{ git_link('jena/src/main/scala/eu/ostrzyciel/jelly/convert/jena/JenaConverterFactory.scala') }})
+    - Example for Jena: [JenaConverterFactory]({{ git_link('jena/src/main/java/eu/neverblink/jelly/convert/jena/JenaConverterFactory.java') }})
+
+
+## Supporting streams
+
+If you want to enable stream support for your library, it will be useful to implement the following utility interfaces from `jelly-core`: `QuadExtractor` and `QuadMaker` for Quads and `TripleExtractor` and `TripleMaker` for Triples. 
```

Comment:
Please change from "streams" to "reactive streams" or "Pekko Stream". A "stream" is a very general and confusing term, be more specific here.